src/com/qamp/HarisJasarevic/homeworks/homework2/task3/Robot.java

```diff
@@ -0,0 +1,74 @@
+package com.qamp.HarisJasarevic.homeworks.homework2.task3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Robot {
+
+    private int x;
+    private int y;
+    private int robotMoves;
+    private List<String> robotPositions;
+
+    public Robot() {
+        this.x = 0;
+        this.y = 0;
+    }
+
+    public Robot(final int x, final int y) {
+        this.x = x;
+        this.y = y;
+        this.robotMoves = 0;
+        this.robotPositions = new ArrayList<>();
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesSouth () {
+        this.y--;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesNorth () {
+        this.y++;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesEast () {
+        this.x++;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public void movesWest () {
+        this.x--;
+        this.robotMoves++;
+        this.robotPositions.add("(" + this.x + "," + this.y + ")");
+    }
+
+    public final String getRobotPosition () {
+        return "(" + this.x + "," + this.y + ")";
+    }
+
+    public final double getRobotDistance () {
+        final double x0 = this.robotPositions.get(0).charAt(1) - '0';
+        final double y0 = this.robotPositions.get(0).charAt(3) - '0';
```

Comment:
This looks a bit ugly, don't you think? But it was really the only way to resolve this in your case.

Except for the looks, the implementation directly depends on the `move` Methods, because if I change the `Strings` this won't work anymore.

All this is the consequence of you storing positions as `String`. Why not create something more fitting, just like Bookshelf stores Books